test/unit/SendDataHandler.spec.ts

```diff
@@ -38,7 +38,7 @@ describe('SendDataHandler', () => {
             await handler.handle(message);
 
             expect(getStatementExecutorSpy.callCount).to.eq(1);
-            expect((writer.ingest as sinon.SinonStub).callCount).to.eq(1);
+            expect((writer.ingest as sinon.SinonStub).callCount).to.eq(2);
```

Comment:
In general, the approach looks awesome. This part seems a little non-intuitive to me, however. We ran one ingestion "task", which resulted in two calls to this function. I'd consider renaming it `upload` or something similar. If I read correctly, an integration config with N queries will produce N+1 uploads (one for each query, and an aggregate one containing the metadata across all queries)